wpilibj/src/test/java/edu/wpi/first/wpilibj/event/BooleanEventTest.java

```diff
@@ -165,4 +241,181 @@ void testEdgeReconstruct() {
 
     assertEquals(4, counter.get());
   }
+
+  /**
+   * Tests that all actions bound to an event will still execute even if the signal is changed
+   * during the loop poll.
+   */
+  @Test
+  void testMidLoopBooleanChange() {
+    var loop = new EventLoop();
+    var bool = new AtomicBoolean(false);
+    var counter = new AtomicInteger(0);
+
+    var event = new BooleanEvent(loop, bool::get).rising();
+    event.ifHigh(
+        () -> {
+          bool.set(false);
+          counter.incrementAndGet();
+        });
+    event.ifHigh(counter::incrementAndGet);
+
+    assertEquals(0, counter.get());
+
+    loop.poll();
+
+    assertEquals(0, counter.get());
+
+    bool.set(true);
+    loop.poll();
+
+    assertEquals(2, counter.get());
+
+    loop.poll();
+
+    assertEquals(2, counter.get());
+
+    bool.set(false);
+    loop.poll();
+
+    assertEquals(2, counter.get());
+
+    bool.set(true);
+    loop.poll();
+
+    assertEquals(4, counter.get());
+  }
+
+  /**
+   * Tests that all actions bound to composed events will still execute even if the composed signal
+   * changes during the loop poll. Also tests that composed edge events only execute on edges (two
+   * rising edge events composed with and() should only execute when both signals are on the rising
+   * edge)
+   */
+  @Test
+  void testMidLoopBooleanChangeWithComposedEvents() {
```

Comment:
This test is too complicated (nearly 100 sloc is a *lot* for a test); it feels like several intertwined test flows, making it very hard to track what's going on.
The `also` in the doc comment also hints at multiple things being tested here.
Can this test be split to multiple simpler tests that each test one clear behavior/case?